torch/_dynamo/variables/torch.py

```diff
@@ -1049,8 +1049,20 @@ def call_function(
 """  # NOQA: B950
                 )
 
+            fn = self.value
+
+            def patched_fn(*args, **kwargs):
+                # This enables reads to global/captured tensors, and we'll just
+                # treat them as constants in the graph. Note that after
+                # AOTDispatcher, this logic would disappear.
+                old_val = fake_tensor_tls.allow_non_fake_inputs_override
+                fake_tensor_tls.allow_non_fake_inputs_override = True
+                res = fn(*args, **kwargs)
+                fake_tensor_tls.allow_non_fake_inputs_override = old_val
```

Comment:
should be try-finally. The case I'm worried about is if `fn` raises an error, the we never reset the fake_tensor_tls

Reply:
Ah right, updated